core/client/fs/src/main/java/alluxio/client/file/dora/WorkerLocationPolicy.java

```diff
@@ -68,11 +71,38 @@ private static class ConsistentHashProvider {
     private List<BlockWorkerInfo> mLastWorkerInfos = ImmutableList.of();
     private NavigableMap<Integer, BlockWorkerInfo> mActiveNodesByConsistentHashing;
 
+    private volatile long mLastUpdatedTimestamp = 0L;
+
+    private final AtomicBoolean mNeedUpdate = new AtomicBoolean(false);
+
+    private static final long WORKER_INFO_UPDATE_INTERVAL_MS = 30_000L;
+
     public void refresh(List<BlockWorkerInfo> workerInfos, int numVirtualNodes) {
-      if (workerInfos != mLastWorkerInfos
-          && !ImmutableSet.copyOf(workerInfos).equals(ImmutableSet.copyOf(mLastWorkerInfos))) {
-        build(workerInfos, numVirtualNodes);
+      // check if we need to update worker info
+      if (mLastUpdatedTimestamp <= 0L
+          || System.currentTimeMillis() - mLastUpdatedTimestamp > WORKER_INFO_UPDATE_INTERVAL_MS) {
+        mNeedUpdate.set(true);
```

Comment:
Does this mean we trade accuracy of the hashing for performance? Maybe leave a todo, so that we can revisit this after we implement a mechanism that notifies the client that there's update to the active worker list.

Reply:
Exactly. We should leave a `TODO` to provide a better way to notify the client.